modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/PostalAddressResourceImpl.java

```diff
@@ -106,6 +130,225 @@ public Page<PostalAddress> getUserAccountPostalAddressesPage(
+			contextCompany.getCompanyId(),
+			contextAcceptLanguage.getPreferredLocale());
+	}
+
+	@Override
+	public PostalAddress postAccountPostalAddress(
+			Long accountId, PostalAddress postalAddress)
+		throws Exception {
+
+		Country country = _getCountryByTitle(postalAddress);
+
+		long regionId = _getRegionByTitleAndCountry(postalAddress, country);
+
+		ListType type = _getType(postalAddress);
+
+		Address address = _addressLocalService.addAddress(
+			null, contextUser.getUserId(), AccountEntry.class.getName(),
+			accountId, postalAddress.getName(), null,
+			postalAddress.getStreetAddressLine1(),
+			postalAddress.getStreetAddressLine2(),
+			postalAddress.getStreetAddressLine3(),
+			postalAddress.getAddressLocality(), postalAddress.getPostalCode(),
+			regionId, country.getCountryId(), type.getListTypeId(), false,
+			postalAddress.getPrimary(), null,
+			ServiceContextFactory.getInstance(contextHttpServletRequest));
+
+		return PostalAddressUtil.toPostalAddress(
+			contextAcceptLanguage.isAcceptAllLanguages(), address,
+			contextCompany.getCompanyId(),
+			contextAcceptLanguage.getPreferredLocale());
+	}
+
+	@Override
+	public PostalAddress putAccountPostalAddress(
+			Long accountId, Long postalAddressId, PostalAddress postalAddress)
+		throws Exception {
+
+		Address address = _validateAccountPostalAddres(
+			accountId, postalAddressId);
+
+		Country country = _getCountryByTitle(postalAddress);
+
+		long regionId = _getRegionByTitleAndCountry(postalAddress, country);
+
+		ListType type = _getType(postalAddress);
+
+		address = _addressLocalService.updateAddress(
+			address.getAddressId(), postalAddress.getName(),
+			address.getDescription(), postalAddress.getStreetAddressLine1(),
+			postalAddress.getStreetAddressLine2(),
+			postalAddress.getStreetAddressLine3(),
+			postalAddress.getAddressLocality(), postalAddress.getPostalCode(),
+			regionId, country.getCountryId(), type.getListTypeId(),
+			address.isMailing(), postalAddress.getPrimary(),
+			address.getPhoneNumber());
+
+		return PostalAddressUtil.toPostalAddress(
+			contextAcceptLanguage.isAcceptAllLanguages(), address,
+			contextCompany.getCompanyId(),
+			contextAcceptLanguage.getPreferredLocale());
+	}
+
+	private Country _getCountryByTitle(PostalAddress postalAddress) {
+		List<Country> countries = _countryService.getCompanyCountries(
+			contextCompany.getCompanyId());
+
+		Iterator<Country> countryIterator = countries.iterator();
+
+		Boolean found = false;
+
+		String title = null;
+
+		Country country = null;
+
+		while (countryIterator.hasNext() && !found) {
+			country = countryIterator.next();
+
+			title = country.getTitle(
+				contextAcceptLanguage.getPreferredLocale());
+
+			if (title.equals(postalAddress.getAddressCountry())) {
+				found = true;
+			}
+		}
+
+		if (!found) {
+			throw new BadRequestException("Country not found");
+		}
+
+		return country;
+	}
+
+	private long _getRegionByTitleAndCountry(
```

Comment:
Should be `_getRegionIdByTitleAndCountry`. I would send it like that, but I suspect Brian will want to rename to  `_getRegionId` because the parameters describe the "by" part.

Reply:
Changed! ([commit](https://github.com/liferay-appsec/liferay-portal/pull/1415/commits/9345dc1fe6935e91140a3f8f58c77406bdad1e91))